textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/FuzzyTokenizer.java

```diff
@@ -0,0 +1,144 @@
+package edu.uci.ics.textdb.dataflow.queryrewriter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by shiladityasen on 4/30/16.
+ * Brute force algorithm to take input a phrase string and output an array of strings as rewritten phrases
+ * A phrase is a string that may comprise of multiple space separated strings called terms
+ * Algorithm tries to find a valid prefix of a term and parses the rest of the term recursively
+ * After recursive call returns with a set of parsed strings, prefix is appended in front of each string
+ */
+public class FuzzyTokenizer
+{
+    //Data members
+    private static WordBase wordBase;
+    private String phrase;
+
+    static {
+        try {
+            wordBase = new WordBase("wordsEn.txt");
+        }
+        catch(IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Parameterized constructor requires phrase string that is to be tokenized
+     * Creates wordBase object containing knowledge base of words in file wordsEn.txt
+     * @param phrase
+     * @throws IOException
+     */
+    public FuzzyTokenizer(String phrase) {
+        this.phrase = phrase;
+    }
+
+    /**
+     * When called returns array of rewritten phrases
+     * Splits phrase string into space separated terms
+     * For each term performs fuzzy tokenization to create corresponding lists of rewritten terms
+     * Calls method rewriteTerm to perform tokenization over single term
+     * Performs a cross product concatenation of each rewritten term list and returns final String array
+     * Calls method crossCatenate to perform required concatenation of multiple lists
+     * @return
+     */
+    public List<String> getFuzzyTokens() {
+        String[] queryWords = phrase.split("\\s+");
```

Comment:
Add a comment: "By default, we use space as a word delimiter."  In the future, we may need to pass an analyzer to do tokenization.